buildcontext/gitlookup.go

```diff
@@ -306,22 +306,26 @@ func (gl *GitLookup) getHostKeyAlgorithms(hostname string) ([]string, []string,
 	if err != nil {
 		gl.console.Warnf("failed to load ~/.ssh/known_hosts: %s", err)
 	}
+	gl.console.VerbosePrintf("loaded %d key(s) from known_hosts and %d default key(s)", len(knownHostsKeyScans), len(defaultKeyScans))
+
+	foundKeys := make(map[string]bool)
 	for _, keyScans := range [][]string{
 		knownHostsKeyScans,
 		defaultKeyScans,
 	} {
 		for _, keyScan := range keyScans {
 			keyAlg, keyData, err := parseKeyScanIfHostMatches(keyScan, hostname)
-			switch err {
-			case nil:
-			case errKeyScanNoMatch:
-				continue
-			default:
-				gl.console.Warnf("failed to parse key scan %q: %s", keyScan, err)
+			if err != nil {
+				gl.console.Warnf("failed to parse key scan %q: %s; skipping", keyScan, err)
 				continue
 			}
 			foundAlgs[keyAlg] = true
-			keys = append(keys, fmt.Sprintf("%s %s %s", knownhosts.Normalize(hostname), keyAlg, keyData))
+			key := fmt.Sprintf("%s %s %s", knownhosts.Normalize(hostname), keyAlg, keyData)
+			if !foundKeys[key] {
+				gl.console.VerbosePrintf("found (normalized) key %s", key)
```

Comment:
nice use of map to reduce messages :)

Reply:
actually, I just realized, if I'm reducing it for the sake of the verbose print, shouldn't I reduce it for the sake of adding it to the keys slice 🤔 ?

Reply:
yeah.... maybe that dedup should be done inside `loadKnownHosts`..... that's a good point :)

Reply:
so I added a dedup step when loading from the file
but I kept trying to dedup again when iterating because some entries will have a hashed host and some won't
so dedup needs to happen after normalizing the host.